### PR TITLE
spelling: add a new hack/verify-spelling script

### PR DESCRIPTION
**Pull request by neolit123**
- pull and build misspell in a temporary dir and run tests on the
kind tree, while skipping vendor.
- update verify-all to run this test first, and move the govet test
before verify-generated.sh.

/kind feature
/area testing
/priority important-longterm
/hold

### CODE REVIEW

**Comment by BenTheElder** [comment]
we have a lot of typos 😁

**Comment by neolit123** [comment]
PR updated.
- made it possible to highlight each error line from misspell in e2e logs (by adding "error: ").
- added spelling fixes.

File: hack/verify-spelling.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+TMP_DIR=$(mktemp -d)
+
+# cleanup
+exitHandler() (
+  echo "Cleaning up ..."
+  rm -rf "${TMP_DIR}"
+)
+trap exitHandler EXIT
+
+# pull misspell
```
Comment (BenTheElder) [inline hack/verify-spelling.sh]: why are we pulling it?

Comment (neolit123) [reply hack/verify-spelling.sh]: should it be vendored - can you elaborate?

also this is still broken for me:
https://github.com/kubernetes/test-infra/blob/master/hack/verify-spelling.sh

Comment (BenTheElder) [reply hack/verify-spelling.sh]: hmm I guess we can not vendor it, I feel like pulling will be flakier in CI. maybe what we should do is have a podutils job run an image with a misspell binary in it.

Comment (BenTheElder) [reply hack/verify-spelling.sh]: vendoring it is worse for the `go get` users that are not developing kind 🤔

Comment (BenTheElder) [reply hack/verify-spelling.sh]: let's go with this for now and revisit later

**Pull request merged**